File: Test7/distinct_lines.pl
#!/usr/bin/perl -w

$counter = 0;
while ($line = <STDIN>){
	$counter++;
	$line = lc($line);
	$line =~ s/\s+/ /g;
	$line =~ s/^ //g;
	$line =~ s/ $//g;
	if(!contained($line, @array)){
		push @array, $line;
	}
	if(@array == $ARGV[0]){
		printf ("%d distinct lines seen after %d lines read.\n", $ARGV[0], $counter);
		exit;
	}
}

print "End of input reached after $counter lines read - $ARGV[0] different lines not seen.\n";

sub contained {
	my($tester, @data) = @_;
	foreach $element (@data){
		if($element eq $tester){
			return 1;
		}
	}
	return 0;
}

File: Test9/distinct_lines.py
#!/usr/bin/python3

import sys, re

counter = 0
array = []
for line in sys.stdin:
	counter += 1
	line = re.sub(' +', ' ', line.lower()).strip()
	#print(line)
	if line in array: continue
	array.append(line)
	if len(array) == int(sys.argv[1]): 
		print("%d distinct lines seen after %d lines read." %(int(sys.argv[1]), counter))
		exit()
print("End of input reached after %d lines read -  %d different lines not seen." %(counter, int(sys.argv[1])))
